sdk/go/src/sawtooth_sdk/client/encoder.go

```diff
@@ -0,0 +1,156 @@
+package client
+
+import (
+	"encoding/hex"
+	"fmt"
+	"github.com/golang/protobuf/proto"
+	"sawtooth_sdk/protobuf/batch_pb2"
+	"sawtooth_sdk/protobuf/transaction_pb2"
+	"strings"
+	"time"
+)
+
+// See sdk/go/tests/client_test.go for example usage.
+
+type Transaction struct{ pb *transaction_pb2.Transaction }
+type Batch struct{ pb *batch_pb2.Batch }
+
+type TransactionEncoder struct {
+	privkey         []byte
+	pubkey          []byte
+	familyName      string
+	familyVersion   string
+	payloadEncoding string
+	addNonce        bool
```

Comment:
Making auto-generated nonce's optional is a good idea. Do you need a boolean flag for it though? Why not just make it so if a nonce is set during transaction creation it is used, otherwise one is auto-generated?